test/e2e/util/util.go

```diff
@@ -1549,3 +1550,140 @@ func validateHostedClusterConditions(t *testing.T, ctx context.Context, client c
+		controlPlaneLabelTolerationKey := "hypershift.openshift.io/control-plane"
+		clusterNodeSchedulingAffinityWeight := 100
+		controlPlaneNodeSchedulingAffinityWeight := clusterNodeSchedulingAffinityWeight / 2
+		colocationLabelKey := "hypershift.openshift.io/hosted-control-plane"
+
+		var podList corev1.PodList
+		if err := client.List(ctx, &podList, crclient.InNamespace(namespace)); err != nil {
+			t.Fatalf("failed to list pods in namespace %s: %v", namespace, err)
+		}
+
+		namespacedName := types.NamespacedName{
+			Namespace: namespace,
+			Name:      hostedCluster.Name,
+		}
+
+		var hcp hyperv1.HostedControlPlane
+		if err := client.Get(ctx, namespacedName, &hcp); err != nil {
+			t.Fatalf("failed to get hostedcontrolplane: %v", err)
+		}
+
+		expected := suppconfig.DeploymentConfig{
+			Scheduling: suppconfig.Scheduling{
+				Tolerations: []corev1.Toleration{
+					{
+						Key:      controlPlaneLabelTolerationKey,
+						Operator: corev1.TolerationOpEqual,
+						Value:    "true",
+						Effect:   corev1.TaintEffectNoSchedule,
+					},
+					{
+						Key:      hyperv1.HostedClusterLabel,
+						Operator: corev1.TolerationOpEqual,
+						Value:    hcp.Namespace,
+						Effect:   corev1.TaintEffectNoSchedule,
+					},
+				},
+				Affinity: &corev1.Affinity{
+					NodeAffinity: &corev1.NodeAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []corev1.PreferredSchedulingTerm{
+							{
+								Weight: int32(controlPlaneNodeSchedulingAffinityWeight),
+								Preference: corev1.NodeSelectorTerm{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      controlPlaneLabelTolerationKey,
+											Operator: corev1.NodeSelectorOpIn,
+											Values:   []string{"true"},
+										},
+									},
+								},
+							},
+							{
+								Weight: int32(clusterNodeSchedulingAffinityWeight),
+								Preference: corev1.NodeSelectorTerm{
+									MatchExpressions: []corev1.NodeSelectorRequirement{
+										{
+											Key:      hyperv1.HostedClusterLabel,
+											Operator: corev1.NodeSelectorOpIn,
+											Values:   []string{hcp.Namespace},
+										},
+									},
+								},
+							},
+						},
+					},
+					PodAffinity: &corev1.PodAffinity{
+						PreferredDuringSchedulingIgnoredDuringExecution: []corev1.WeightedPodAffinityTerm{
+							{
+								Weight: 100,
+								PodAffinityTerm: corev1.PodAffinityTerm{
+									LabelSelector: &metav1.LabelSelector{
+										MatchLabels: map[string]string{
+											colocationLabelKey: hcp.Namespace,
+										},
+									},
+									TopologyKey: corev1.LabelHostname,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		for _, pod := range podList.Items {
+			for _, expectedToleration := range expected.Scheduling.Tolerations {
+				found := false
+				for _, podToleration := range pod.Spec.Tolerations {
+					if reflect.DeepEqual(podToleration, expectedToleration) {
```

Comment:
instead of this loops+if stanzas can we just do as in testSetLocation, e.g.
`g.Expect(expected.Scheduling.Tolerations).To(BeEquivalentTo(cfg.Scheduling.Tolerations))`

You might also want to leverage other helpers like ContainElements(
`g.Expect(cfg.Scheduling.Tolerations).To(ContainElements(expected.Scheduling.Tolerations))`

Reply:
Ah wasnt aware of containElements that makes it much easier.

Reply:
was about to comment the same :) `g.ContainElements` will make it much more cleaner.